state-sync/src/coordinator.rs

```diff
@@ -199,69 +198,116 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
+                            }
                         }
-                        Event::Message(peer_id, message) => self.process_one_message(PeerNetworkId(network_id.clone(), peer_id), message).await,
                         unexpected_event => {
                             counters::NETWORK_ERROR_COUNT.inc();
                             warn!(LogSchema::new(LogEntry::NetworkError),
                             "received unexpected network event: {:?}", unexpected_event);
                         },
+
                     }
                 },
                 _ = interval.select_next_some() => {
-                    self.check_progress();
+                    match self.check_progress() {
+                        Ok(()) => {},
+                        Err(Error::ConsensusIsExecuting) => { /* Don't log to avoid spamming logs! */ },
+                        Err(error) => {
+                            error!(LogSchema::event_log(LogEntry::ProgressCheck, LogEvent::Fail).error(&error.into()));
+                        }
+                    }
                 }
             }
         }
     }
 
-    pub(crate) async fn process_one_message(&mut self, peer: PeerNetworkId, msg: StateSyncMessage) {
+    fn process_new_peer(
+        &mut self,
+        network_id: NodeNetworkId,
+        peer_id: PeerId,
+        origin: ConnectionOrigin,
+    ) -> Result<(), Error> {
+        let peer = PeerNetworkId(network_id, peer_id);
+        self.request_manager.enable_peer(peer, origin)?;
+        self.check_progress()
+    }
+
+    fn process_lost_peer(
+        &mut self,
+        network_id: NodeNetworkId,
+        peer_id: PeerId,
+        origin: ConnectionOrigin,
+    ) -> Result<(), Error> {
+        let peer = PeerNetworkId(network_id, peer_id);
+        self.request_manager.disable_peer(&peer, origin)
+    }
+
+    pub(crate) async fn process_chunk_message(
+        &mut self,
+        network_id: NodeNetworkId,
+        peer_id: PeerId,
+        msg: StateSyncMessage,
+    ) -> Result<(), Error> {
+        let peer = PeerNetworkId(network_id, peer_id);
         match msg {
             StateSyncMessage::GetChunkRequest(request) => {
+                // Time message handling
                 let _timer = counters::PROCESS_MSG_LATENCY
                     .with_label_values(&[
                         &peer.raw_network_id().to_string(),
                         &peer.peer_id().to_string(),
                         counters::CHUNK_REQUEST_MSG_LABEL,
                     ])
                     .start_timer();
-                let result_label =
-                    if let Err(err) = self.process_chunk_request(peer.clone(), *request.clone()) {
+
+                // Process chunk request
+                let result_label = match self.process_chunk_request(peer.clone(), *request.clone())
+                {
+                    Ok(()) => counters::SUCCESS_LABEL,
+                    Err(error) => {
```

Comment:
Curious, why the switch?  This is the same logic as an `if let` in the `match` with no extra information.

Reply:
Personal preference, nothing more! 😄